h5py/_hl/files.py

```diff
@@ -209,7 +209,18 @@ def make_fid(name, mode, userblock_size, fapl, fcpl=None, swmr=False):
         # existing one (ACC_EXCL)
         try:
             fid = h5f.open(name, h5f.ACC_RDWR, fapl=fapl)
-        except FileNotFoundError:
+        # Not all drivers raise FileNotFoundError (commented those that do not)
+        except FileNotFoundError if fapl.get_driver() in (
```

Comment:
I am 50/50 that it is worth the complexity to make the exception slightly stricter.

Reply:
There have been a few issues filed through the years where the overly broad `except` obscured the original, meaningful error. At a quick look, that includes permission errors (#1901), file corruption (#940) and locking (#712, #780). Since the default driver does give us an errno, I'm convinced that it's worth dealing with some extra complexity to be specific when possible.

If you don't like the hardcoded list of drivers, another way we could do this is to check the exception, and if there's no errno, assume it's probably FileNotFound (like we assumed all OSErrors were before 3.3):

```python
except OSError as e:
    if (e.errno is not None) and (e.errno != errno.ENOENT):
        raise
    fid = h5f.create(name, h5f.ACC_EXCL, fapl=fapl, fcpl=fcpl)
```